### PR TITLE
DOCS-2419: Merge autogen: arm

### PR DESCRIPTION
**Pull request by andf-viam**
Merge `arm` component generated content.
- Follow the wiki page for guidance on how to do this. This arm PR to serve as example 'template' for the rest of them. See https://viam.atlassian.net/browse/DOCS-2365.

Note:
- This still bears an issue where Viam-specific data types, like `spatialmath.Pose`, appear in returns with bogus return names. I'll fix that shortly in a follow-on.
- This still includes a few links to SDK docs in the proto descriptions. I'll remove those also in a follow-on shortly.

### CODE REVIEW

File: static/include/components/apis/generated/arm-table.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!-- prettier-ignore -->
+Method Name | Description
+----------- | -----------
+[`GetEndPosition`](/components/arm/#getendposition) | Get the current position of the arm as a [Pose](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.Pose).
```
Comment (npentrel) [inline static/include/components/apis/generated/arm-table.md:4]: I don't this this should link to the python Pose class. Maybe we need to remove links from descriptions if that's the only way

Comment (npentrel) [reply static/include/components/apis/generated/arm-table.md:4]: Not a merge blocker. you can file a ticket to fix this later

File: static/include/components/apis/generated/arm.md [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- `extra` (Mapping[[str](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str), Any]) (optional): Extra options to pass to the underlying RPC call.
+- `timeout` ([float](https://docs.python.org/3/library/stdtypes.html#numeric-types-int-float-complex)) (optional): An option to set how long to wait (in seconds) before calling a time-out and closing the underlying RPC call.
+
+**Returns:**
+
+- ([viam.proto.component.arm.JointPositions](https://python.viam.dev/autoapi/viam/proto/component/arm/index.html#viam.proto.component.arm.JointPositions)): The current JointPositions for the arm.
```
Comment (npentrel) [inline static/include/components/apis/generated/arm.md:169]:
```suggestion
- ([viam.proto.component.arm.JointPositions](https://python.viam.dev/autoapi/viam/proto/component/arm/index.html#viam.proto.component.arm.JointPositions)): The current JointPositions of the arm.
```

**Comment by npentrel** [comment]
This looks great, once the code snippets are in (which should be quick if it only requires running locally?), I think this should merge and we can address the descriptions afterwards in another ticket

File: static/include/components/apis/generated/arm.md (hunk 2 above)
Comment (npentrel) [reply static/include/components/apis/generated/arm.md:169]: not a merge blocker but would be good to fix upstream

File: static/include/components/apis/generated/arm-table.md (hunk 1 above)
Comment (andf-viam) [reply static/include/components/apis/generated/arm-table.md:4]: Now stripping links from these MD table files in https://github.com/viamrobotics/docs/pull/2984!

File: docs/build/program/apis/_index.md [hunk 3]
```diff
@@ -61,7 +61,7 @@ Documentation on using these methods in your SDK code is found on each [componen
 
 The [arm component](/components/arm/) supports the following methods:
 
-{{< readfile "/static/include/components/apis/arm.md" >}}
```
[the remark this answers is not in the document]
Comment (andf-viam) [reply docs/build/program/apis/_index.md:64]: You've convinced me, I'll delete the original include as part of this PR as well. 👍

File: static/include/components/apis/generated/arm.md [hunk 4]
```diff
@@ -0,0 +1,536 @@
...
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/resource#Actuator).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### GetGeometries
+
+Get all the geometries associated with the arm in its current configuration, in the [frame](/mobility/frame-system/) of the arm.
```
Comment (npentrel) [inline static/include/components/apis/generated/arm.md:386]: I think this one is a problem, is that an overwrite?

**Review by npentrel — APPROVED** [review]
LGTM % link failures

**Pull request merged**